src/dataServices/apiDataService.ts

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable class-methods-use-this */
+
+import {
+    ConditionalDeleteResourceRequest,
+    CreateResourceRequest,
+    DeleteResourceRequest,
+    GenericResponse,
+    PatchResourceRequest,
+    Persistence,
+    ReadResourceRequest,
+    ResourceNotFoundError,
+    UpdateResourceRequest,
+    vReadResourceRequest,
+} from 'fhir-works-on-aws-interface';
+import axios from 'axios';
+
+export class ApiDataService implements Persistence {
+    updateCreateSupported: boolean = false;
+
+    // TODO: Pull this value from AWS Param Store
+    private INTEGRATION_TRANSFORM_URL: string = '';
+
+    constructor(integrationTransformUrl: string) {
+        this.INTEGRATION_TRANSFORM_URL = integrationTransformUrl;
+    }
+
+    async createResource(request: CreateResourceRequest): Promise<GenericResponse> {
+        try {
+            const url = `${this.INTEGRATION_TRANSFORM_URL}/${request.resourceType}`;
+            const response = await axios.post(url, request.resource);
+            return { message: '', resource: response.data.resource };
+        } catch (e) {
+            throw this.getError(e);
+        }
+    }
+
+    async readResource(request: ReadResourceRequest): Promise<GenericResponse> {
+        try {
+            const response = await axios.get(`${this.INTEGRATION_TRANSFORM_URL}/${request.resourceType}/${request.id}`);
+            return { message: '', resource: response.data.resource };
+        } catch (e) {
+            throw this.getError(e, request.resourceType, request.id);
+        }
+    }
+
+    async updateResource(request: UpdateResourceRequest): Promise<GenericResponse> {
+        try {
+            const response = await axios.put(
+                `${this.INTEGRATION_TRANSFORM_URL}/${request.resourceType}/${request.id}`,
+                request.resource,
+            );
+            return { message: '', resource: response.data.resource };
+        } catch (e) {
+            throw this.getError(e, request.resourceType, request.id);
+        }
+    }
+
+    async deleteResource(request: DeleteResourceRequest): Promise<GenericResponse> {
+        try {
+            await axios.delete(`${this.INTEGRATION_TRANSFORM_URL}/${request.resourceType}/${request.id}`);
+            // Don't need to actually return anything to the router
+            return { message: '', resource: {} };
```

Comment:
resource is optional. I think it's better to just omit it
```suggestion
            return { message: ''};
```

Reply:
updated